File: android/app/src/main/kotlin/pt/ualg/miaugenda/data/remote/AttendanceApi.kt
package pt.ualg.miaugenda.data.remote

import pt.ualg.miaugenda.data.model.AttendanceRecord
import pt.ualg.miaugenda.data.model.CreateAttendanceBody
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface AttendanceApi {
    @POST("/attendance")
    suspend fun register(
        @Body body: CreateAttendanceBody
    ): Response<AttendanceRecord>

    @GET("/attendance/me")
    suspend fun getMyHistory(
        @Query("from") from: String? = null,
        @Query("to") to: String? = null
    ): Response<List<AttendanceRecord>>

    @GET("/attendance")
    suspend fun getAttendance(
        @Query("userId") userId: Int? = null,
        @Query("from") from: String? = null,
        @Query("to") to: String? = null
    ): Response<List<AttendanceRecord>>
}

File: desktop/src/api/attendanceApi.ts
import { apiClient } from "./apiClient";
import type {
  ActiveEmployee,
  AttendanceRecord,
  CreateAttendanceBody
} from "../models/attendance";

export interface AttendanceFilters {
  userId?: number | null;
  from?: string | null;
  to?: string | null;
}

export async function registerAttendance(
  body: CreateAttendanceBody
): Promise<AttendanceRecord> {
  const response = await apiClient.post<AttendanceRecord>("/attendance", body);
  return response.data;
}

export async function getMyAttendanceHistory(
  filters: Omit<AttendanceFilters, "userId"> = {}
): Promise<AttendanceRecord[]> {
  const response = await apiClient.get<AttendanceRecord[]>("/attendance/me", {
    params: {
      from: filters.from ?? undefined,
      to: filters.to ?? undefined
    }
  });

  return response.data;
}

export async function getAttendance(
  filters: AttendanceFilters = {}
): Promise<AttendanceRecord[]> {
  const response = await apiClient.get<AttendanceRecord[]>("/attendance", {
    params: {
      userId: filters.userId ?? undefined,
      from: filters.from ?? undefined,
      to: filters.to ?? undefined
    }
  });

  return response.data;
}

export async function getActiveEmployees(): Promise<ActiveEmployee[]> {
  const response = await apiClient.get<ActiveEmployee[]>("/attendance/active");
  return response.data;
}